File: PTApi/Data/Models/Mahasiswa.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PTApi.Data.Models
{
    public class Mahasiswa
    {
        [Key]
        public int Nim { get; set; }
        public string Nama { get; set; }
        [Column(TypeName = "date")]
        public DateTime TgLahir { get; set; }
        public string TpLahir { get; set; }
        public string JenisKelamin { get; set; }
        public List<MhsKls> Klasses { get; set; }
    }
}

File: PTUI/src/app/model/mahasiswa.ts
import { Kelas } from '../model/kelas';

export class Mahasiswa {
    constructor(
        public Nim: number,
        public Nama: string,
        public TgLahir: string,
        public TpLahir: string,
        public JenisKelamin: string,
        public Klasses: Kelas[],
        public NilaiMid: string,
        public NilaiSem: string,
        public NilaiMutu: string
    ) { }
}